### PR TITLE
simplify banning and broadcasting

### LINKED ISSUE #57: simplify banning and broadcast logic
1. actually remove player when they are banned from a pool. this simplifies broadcast testing
2. iterate over the players in the pool rather than all connections on the tracker when making a broadcast.

@zquestz I feel like there might be a reason that we didn't do 1. already but I can't think of it. Any problem with removing players immediately on ban?

### PR DESCRIPTION
**Pull request by emergent-reasons**
Fixes #57

- single ban, no voter tracking
- more efficient and less complex looping over pool players
- remove redundant broadcast return errors
- remove redundant tests for banned player since banned is remove immediately

### CODE REVIEW

**Comment by emergent-reasons** [comment]
I am not especially excited about this one. If the value of changes is minimal, maybe it's better to just leave it. It's just part of what is possible on simplifying the pool logic since we assume there are no loops in player/pool state any more.

File: server/broadcast.go [hunk 1]
```diff
@@ -69,21 +61,13 @@ func (pi *packetInfo) broadcastAll(msgs []*message.Signed) error {
...
 	// the broadcast.
 	if sender == nil {
-		return nil
+		return
 	}
 
-	for conn, player := range pi.tracker.connections {
-		if (sender.pool != player.pool) || sender.pool.IsBanned(player) {
-			continue
-		}
```
Comment (emergent-reasons) [inline server/broadcast.go]: Both of these conditions should be redundant now.

File: server/broadcast.go [hunk 2]
```diff
@@ -100,36 +84,29 @@ func (pi *packetInfo) announceStart() {
 		return
 	}
 
-	for conn, player := range pi.tracker.connections {
-		if sender.pool != player.pool || sender.pool.IsBanned(player) {
-			continue
-		}
-
```
Comment (emergent-reasons) [inline server/broadcast.go]: same here

File: server/verification.go [hunk 3]
```diff
@@ -22,10 +22,6 @@ func (pi *packetInfo) verifyMessage() error {
 			return errors.New("invalid verification key")
 		}
 
-		if player.pool.IsBanned(player) {
-			return errors.New("banned player")
-		}
-
```
Comment (emergent-reasons) [inline server/verification.go]: redundant as above

**Comment by zquestz** [comment]
I actually think these changes are quite good.

1) The code is much simpler.
2) We removed poolVoters which isn't needed.
3) The broadcasts now only loop through pool participants.

Going to merge. =)

**Pull request merged**